includes/embeds/class-amp-soundcloud-embed.php

```diff
@@ -33,21 +60,81 @@ public function get_scripts() {
 		return array( self::$script_slug => self::$script_src );
 	}
 
-	public function oembed( $matches, $attr, $url, $rawattr ) {
+	/**
+	 * Render oEmbed.
+	 *
+	 * @see \WP_Embed::shortcode()
+	 *
+	 * @deprecated Core's oEmbed handler is now used instead, with embed_oembed_html filter used to convert to AMP.
+	 * @param array  $matches URL pattern matches.
+	 * @param array  $attr    Shortcode attribues.
+	 * @param string $url     URL.
+	 * @return string Rendered oEmbed.
+	 */
+	public function oembed( $matches, $attr, $url ) {
+		_deprecated_function( __METHOD__, '0.6.0' );
+		unset( $matches, $attr );
 		$track_id = $this->get_track_id_from_url( $url );
-		return $this->render( array(
-			'track_id' => $track_id,
-		) );
+		return $this->render( compact( 'track_id' ) );
 	}
 
-	public function shortcode( $attr ) {
+	/**
+	 * Filter oEmbed HTML for SoundCloud to convert to AMP.
+	 *
+	 * @param string $cache Cache for oEmbed.
+	 * @param string $url   Embed URL.
+	 * @return string Embed.
+	 */
+	public function filter_embed_oembed_html( $cache, $url ) {
+		$parsed_url = wp_parse_url( $url );
+		if ( $url && false === strpos( $parsed_url['host'], 'soundcloud.com' ) ) {
```

Comment:
While the risk is minimal, this could throw a PHP warning if the url passed doesn't have an host.

Reply:
I don't believe this can ever happen.